tests/integration/conftest.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+import pexpect
+
+class HelperBase:
+
+    def __dumptext(self):
+        print("Debug help: actual output---------------------")
+        print(self.pexpectobj.before)
+        print("----------------------------------------------")
+
+    def commandresponse(self, cmd,resp):
+        self.pexpectobj.sendline(cmd)
+        try:
+            self.pexpectobj.expect_exact(resp)
+        except:
+            self.__dumptext()
+            return False
+        return True
+
+    def waitforpacket(self):
+        try:
+            self.pexpectobj.expect_exact("bytes received from")
+        except:
+            self.__dumptext()
+            assert False
+        #save match line
+        try:
+            self.pexpectobj.expect_exact("\r\r\n>")
+        except:
+            __dumptext()
+            assert False
+        return self.pexpectobj.before
+
+    def waitfortext(self,txt):
+        try:
+            self.pexpectobj.expect_exact(txt)
+        except:
+            self.__dumptext()
+            return False
+        return True
+
+    def quit(self):
+            self.pexpectobj.sendline("q")
+            self.pexpectobj.sendline("quit")
+            self.pexpectobj.expect(pexpect.EOF)
+
+
+class Lwm2mServer(HelperBase):
+    def __init__(self, arguments="", timeout=3, encoding="utf8"):
+        self.pexpectobj = pexpect.spawn("build-wakaama/examples/server/lwm2mserver" + " " + arguments,
+                               encoding=encoding,
+                               timeout=timeout)
+        # uncomment to enable logging, helpful when debugging
+        # self.pexpectobj.logfile = open("lwm2mserver_log.txt", "w")
+
+class Lwm2mClient(HelperBase):
+    def __init__(self, arguments="", timeout=3, encoding="utf8"):
+        self.pexpectobj = pexpect.spawn("build-wakaama/examples/client/lwm2mclient" + " " + arguments,
+                               encoding=encoding,
+                               timeout=timeout)
+        # uncomment to enable logging, helpful when debugging
+        # self.pexpectobj.logfile = open("lwm2mclient_log.txt", "w")
+
+
+class Lwm2mBootstrapServer(HelperBase):
+    def __init__(self, arguments="", timeout=3, encoding="utf8"):
+        self.pexpectobj = pexpect.spawn("build-wakaama/examples/bootstrap_server/bootstrap_server" + " " + arguments,
+                               encoding=encoding,
+                               timeout=timeout)
+        # uncomment to enable logging, helpful when debugging
+        # self.pexpectobj.logfile = open("lwm2mbootstrapserver_log.txt", "w")
+
+
+@pytest.fixture
+def lwm2mserver():
+    """Provide lwm2mserver instance."""
+    server = Lwm2mServer()
+    yield server
+    server.quit()
+
+
+@pytest.fixture
+def lwm2mclient():
```

Comment:
Documentation should make clear what the difference to `lwm2mclient_boot` is.

Reply:
It is the command line arguments you pass. Besides documenting I'm really interested in a nicer way to do it

Reply:
this has been refactored in newer version